public/redux/reducers/anomalyResults.ts

```diff
@@ -104,7 +105,7 @@ export const getDetectorResults = (
         type: DETECTOR_RESULTS,
         request: (client: HttpSetup) =>
           client.get(
-            `..${AD_NODE_API.DETECTOR}/${id}/results/${isHistorical}`,
+            `..${AD_NODE_API.DETECTOR}/${id}/${dataSourceId}/results/${isHistorical}`,
```

Comment:
Same deal as `startDetector`. If `dataSourceId = ""`, the path would be `/id//results/...`